tools/donate-cpu-server.py

```diff
@@ -10,10 +10,40 @@
 from threading import Thread
 import sys
 import urllib
+import logging
+import logging.handlers
 
 OLD_VERSION = '1.86'
 
 
+# Set up logging
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+# Logging to console
+handler_stream = logging.StreamHandler()
+logger.addHandler(handler_stream)
+# Log errors to a rotating file
+logfile = sys.path[0]
+if logfile:
+    logfile += '/'
+logfile += 'donate-cpu-server.log'
```

Comment:
can `logfile` be None here? I see the above condition.

Reply:
It should never be `None` but it could theoretically be an empty string.
If the code is executed directly inside the interpreter and not in a file then `sys.path[0]` could be empty because there is no script directory. Then it does not make sense to append the `/`, which would then point to the root directory.
I admit this is very defensive programming :)

Reply:
ok